pkg/generators/extension.go

```diff
@@ -48,24 +63,67 @@ type extension struct {
 	values []string // Example: [atomic]
 }
 
+func (e extension) hasAllowedValues() bool {
+	xAttrs, exists := tagToExtension[e.idlTag]
+	if !exists || xAttrs.allowedValues == nil {
```

Comment:
I'm sure you're more interested in feedback wrt the structure of the code more than this kind of nit, but yet, here I go: this function can be simplified to:
```
func (e extension) hasAllowedValues() bool {
   len(tagToExtension[e.idleTag].allowedValues) > 0
}
```

I'm not sure I'd suggest this extreme form, but you get the idea. Actually yeah, I think the intent is there.